packages/logger-plugin/src/symbols.ts

```diff
@@ -9,6 +9,9 @@ export interface NgxsLoggerPluginOptions {
 
   /** Disable the logger. Useful for prod mode. */
   disabled?: boolean;
+
+  /** Predicate for actions to be the logged. Takes action and state snapshot as parameters */
+  filter?: (action: any, state: any) => boolean;
```

Comment:
```ts
filter?: <T extends PlainObject>(action: ActionType, state: T) => boolean;
```

Reply:
It would have caused a warning in Ivy if I did that, because `PlainObject` is an internal type of the store package.

Reply:
ok

```ts
filter?: (action: ActionType, state: any) => boolean;
```

Reply:
`action` here is not the constructor, but the instance. Therefore, `ActionType` would be misleading. People would think they could access static `type` property (like `action.type`), but they cannot and they will get `undefined` instead of the type.

Reply:
I agree that `ActionType` would not work here, because an action's type could be available directly on the object or as a static property. We would not want to mislead either way.
The user would have to use the `getActionTypeFromInstance` method exposed from `@ngxs/store` in order to safely get the type if that is the information they are after.
I think that we have to stick with the `any` type for both parameters here.